--- gtksourceview-3.24.11/gtksourceview/gtksourceview.c~	2019-03-21 22:55:02.000000000 +0100
+++ gtksourceview-3.24.11/gtksourceview/gtksourceview.c	2025-07-19 22:12:05.184791972 +0200
@@ -1586,7 +1586,7 @@ set_source_buffer (GtkSourceView *view,
 	{
 		GtkSourceBufferInternal *buffer_internal;
 
-		view->priv->source_buffer = g_object_ref (buffer);
+		view->priv->source_buffer = (GtkSourceBuffer *)g_object_ref (buffer);
 
 		g_signal_connect (buffer,
 				  "highlight-updated",
